Objects/unicodeobject.c

```diff
@@ -629,6 +627,39 @@ _PyUnicode_CheckConsistency(PyObject *op, int check_content)
         }
         CHECK(PyUnicode_READ(kind, data, ascii->length) == 0);
     }
+
+    /* Check interning state */
+#ifdef Py_DEBUG
+    switch (PyUnicode_CHECK_INTERNED(op)) {
+        case SSTATE_NOT_INTERNED:
+            if (ascii->state.statically_allocated) {
+                CHECK(_Py_IsImmortal(op));
+                // This state is for two exceptions:
+                // - strings are currently checked before they're interned
```

Comment:
This comment still stands?  At least, `_PyUnicode_CheckConsistency()` now follows `_PyUnicode_InternStatic()` in `generate_global_objects.py`.

Reply:
Yes, but `_PyUnicode_InternStatic` itself starts out with a check. (That could be removed; this PR does have more checks and assertions than needed.)